ci-operator/step-registry/ipi/conf/gcp/aws-route53/ipi-conf-gcp-aws-route53-commands.sh

```diff
@@ -0,0 +1,123 @@
+#!/bin/bash
+
+set -o nounset
+set -o errexit
+set -o pipefail
+
+dns_reserve_and_defer_cleanup() {
+  record=$1
+  hostname=$2
+  recordtype=$3
+
+  json_create="{
+    \"Action\": \"UPSERT\",
+    \"ResourceRecordSet\": {
+      \"Name\": \"$hostname\",
+      \"Type\": \"$recordtype\",
+      \"TTL\": 60,
+      \"ResourceRecords\": [{\"Value\": \"$record\"}]
+    }
+  }"
+  jq --argjson jc "$json_create" '.Changes += [$jc]' "${SHARED_DIR}"/dns-create.json > "${SHARED_DIR}"/tmp-create.json && mv "${SHARED_DIR}"/tmp-create.json "${SHARED_DIR}"/dns-create.json
+
+  json_delete="{
+    \"Action\": \"DELETE\",
+    \"ResourceRecordSet\": {
+      \"Name\": \"$hostname\",
+      \"Type\": \"A\",
+      \"TTL\": 60,
+      \"ResourceRecords\": [{\"Value\": \"$record\"}]
+    }
+  }"
+  jq --argjson jd "$json_delete" '.Changes += [$jd]' "${SHARED_DIR}"/dns-delete.json > "${SHARED_DIR}"/tmp-delete.json && mv "${SHARED_DIR}"/tmp-delete.json "${SHARED_DIR}"/dns-delete.json
+}
+
+function find_out_api_and_ingress_ip_addresses() {
+  local -r infra_id="$1"
+  local -r region="$2"
+  local -r out_file="$3"
+  local api_ip_address ingress_ip_address ingress_forwarding_rule
+
+  api_ip_address=$(gcloud compute forwarding-rules describe --global "${infra_id}-apiserver" --format json | jq -r .IPAddress)
+  ingress_forwarding_rule=$(gcloud compute target-pools list --format=json --filter="instances[]~${infra_id}" | jq -r .[].name)
+  if [[ -n "${ingress_forwarding_rule}" ]]; then
+    ingress_ip_address=$(gcloud compute forwarding-rules describe --region "${region}" "${ingress_forwarding_rule}" --format json | jq -r .IPAddress)
+  else
+    ingress_ip_address=""
+    echo "$(date -u --rfc-3339=seconds) - ERROR: Failed to find the INGRESS forwarding-rule."
+  fi
+
+  echo "$(date -u --rfc-3339=seconds) - INFO: Populate the file '${out_file}' with API server IP (${api_ip_address}) and INGRESS server IP (${ingress_ip_address})..."
+  cat > "${out_file}"  << EOF
+${api_ip_address}
+${ingress_ip_address}
+EOF
+}
+
+AWS_BASE_DOMAIN="qe.devcluster.openshift.com"
+
+cluster_name=${NAMESPACE}-${UNIQUE_HASH}
+cluster_domain="${cluster_name}.${AWS_BASE_DOMAIN}"
+INFRA_ID="$(jq -r .infraID ${SHARED_DIR}/metadata.json)"
+
+export AWS_DEFAULT_REGION=us-west-2  # TODO: Derive this?
```

Comment:
do we really need this `AWS_DEFAULT_REGION` for running the `aws  route53` commands? I remember the Route53 is global resource.

Reply:
I copied it from [here](https://github.com/openshift/release/blob/master/ci-operator/step-registry/ipi/conf/vsphere/dns/ipi-conf-vsphere-dns-commands.sh#L45), not quite sure if it is necessary.